src/commonMain/kotlin/fr/acinq/lightning/channel/Syncing.kt

```diff
@@ -46,15 +50,35 @@ data class Syncing(val state: ChannelStateWithCommitments, val waitForTheirReest
                     } else {
                         state
                     }
-                    val yourLastPerCommitmentSecret = nextState.commitments.remotePerCommitmentSecrets.lastIndex?.let { nextState.commitments.remotePerCommitmentSecrets.getHash(it) } ?: ByteVector32.Zeroes
-                    val myCurrentPerCommitmentPoint = keyManager.commitmentPoint(state.commitments.params.localParams.channelKeys(keyManager).shaSeed, nextState.commitments.localCommitIndex)
-                    val channelReestablish = ChannelReestablish(
-                        channelId = nextState.channelId,
-                        nextLocalCommitmentNumber = nextState.commitments.localCommitIndex + 1,
-                        nextRemoteRevocationNumber = nextState.commitments.remoteCommitIndex,
-                        yourLastCommitmentSecret = PrivateKey(yourLastPerCommitmentSecret),
-                        myCurrentPerCommitmentPoint = myCurrentPerCommitmentPoint
-                    ).withChannelData(nextState.commitments.remoteChannelData, logger)
+                    val channelReestablish = when (nextState) {
```

Comment:
This part should probably be factored between `Offline` and `Syncing`. Maybe we should initialize an `Offline` state here and make it process the `channel_reestablish`?

Reply:
Refactored in https://github.com/ACINQ/lightning-kmp/pull/426/commits/93c214b7cdcde0149f64389590a4ed940fa9a55c, I simply created a helper function in `PersistedChannelState` that create `channel_reestablish`.